handler: fix handler precedence (name > listen)

resmo opened the pull request:
##### ISSUE TYPE

<!--- Pick one below and delete the rest: -->
- Bugfix Pull Request
##### COMPONENT NAME

handler
##### ANSIBLE VERSION

<!--- Paste verbatim output from “ansible --version” between quotes below -->

```
2.2
```
##### SUMMARY

Fixes #17855 
Successfully tested with integration tests from #17070

bcoca commented on lib/ansible/plugins/strategy/__init__.py:
```diff
@@ -395,35 +395,35 @@ def parent_handler_match(target_handler, handler_name):
-                                            raise AnsibleError("The requested handler '%s' was found in neither the main handlers list nor the listening handlers list" % handler_name)
+                                # and if none were found, then we raise a warning
+                                if not handler_found:
+                                    display.warning("The requested handler '%s' was found in neither the main handlers list nor the listening handlers list" % handler_name)
```
This should be an error by default, created #17902 to allow this to be configurable so this change should be removed from here.

resmo commented:
As discussed in IRC precedence is obsolete, both types, listen handles and named handlers should run.

bcoca commented:
order might still matter, i do think 'name' should run before 'listen'

resmo committed:
handler: notify a handler by name _and_ listen

Before we only allowed either notify by name or listen and name had precedence.

resmo commented:
ready_for_review
